core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -435,20 +436,167 @@ ClientRequestContext newDerivedContext(RequestId id, @Nullable HttpRequest req,
+     * assert ctx.responseTimeoutMillis() == oldResponseTimeoutMillis + 500;
+     * }</pre>
+     *
+     * @param adjustmentMillis the amount of time in milliseconds to extend the existing timeout
+     */
+    void extendResponseTimeoutMillis(long adjustmentMillis);
+
+    /**
+     * Extends the existing response timeout allowed by the specified {@code adjustment} until
+     * since the transfer of the {@link Response} started.
+     * Note that the negative {@code adjustment} reduces the existing timeout.
+     * The initial timeout is set from {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * <p>For example:
+     * <pre>{@code
+     * ClientRequestContext ctx = ...;
+     * long oldResponseTimeoutMillis = ctx.responseTimeoutMillis();
+     * ctx.extendResponseTimeout(Duration.ofSeconds(1));
+     * assert ctx.responseTimeoutMillis() == oldResponseTimeoutMillis + 1000;
+     * ctx.extendResponseTimeout(Duration.ofMillis(-500));
+     * assert ctx.responseTimeoutMillis() == oldResponseTimeoutMillis + 500;
+     * }</pre>
+     *
+     * @param adjustment the amount of time to extend the existing timeout
+     */
+    void extendResponseTimeout(Duration adjustment);
+
+    /**
+     * Sets the amount of time allowed that is after the specified {@code responseTimeoutMillis}
+     * from now until receiving the {@link Response} completely since the transfer
+     * of the {@link Response} started.
+     * Note that the specified {@code responseTimeoutMillis} must be positive.
+     * The initial timeout is set from {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * <p>For example:
+     * <pre>{@code
+     * ClientRequestContext ctx = ...;
+     * // Set the deadline to 'now' + 1000
+     * ctx.setResponseTimeoutAfterMillis(1000);
+     * }</pre>
+     *
+     * @param responseTimeoutMillis the amount of time allowed in milliseconds from now
+     */
+    void setResponseTimeoutAfterMillis(long responseTimeoutMillis);
+
+    /**
+     * Sets the amount of time allowed that is after the specified {@code responseTimeout}
+     * from now until receiving the {@link Response} completely since the transfer
+     * of the {@link Response} started.
+     * Note that the specified {@code responseTimeout} must be positive.
+     * The initial timeout is set from {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * <p>For example:
+     * <pre>{@code
+     * ClientRequestContext ctx = ...;
+     * // Set the deadline to 'now' + Duration.ofSeconds(1)
+     * ctx.setResponseTimeoutAfter(Duration.ofSeconds(1));
+     * }</pre>
+     *
+     * @param responseTimeout the amount of time allowed from now
+     */
+    void setResponseTimeoutAfter(Duration responseTimeout);
+
+    /**
+     * Sets the absolute response timeout allowed in milliseconds from the epoch of 1970-01-01T00:00:00Z
+     * until receiving the {@link Response} completely since the transfer of the {@link Response} started.
```

Comment:
How about this:

```
Schedules the response timeout that is triggered at the specified time represented
as the number of milliseconds since the epoch ({@code 1970-01-01T00:00:00Z}).
```